### ISSUE TITLE
FactoryGirl shouldn't patch core classes

### ISSUE THREAD

**Issue by rosenfeld** [thoughtbot/factory_bot#1006]
Hi, I removed Rails from a 5 years-old project and I can't feel enough confidence that it should work completely because the application doesn't depend on ActiveSupport but FactoryGirl not only depends on ActiveSupport, but worse than that, it depends on some monkey patches to core Ruby classes.

That means my specs could pass and the real production code could fail because it would be using some of those monkey patches that would only be present in the test environment.

Ideally it shouldn't depend on activesupport for anything so that if the production code tries to rely on ActiveSupport and it's not present, tests should fail as expected.

Does that make sense?

**Comment by davydovanton** [comment]
Рey @rosenfeld. You need to check this issue :)
https://github.com/thoughtbot/factory_girl/issues/903

Also, if you want to use factory gem without AS I can suggest https://www.fabricationgem.org or https://github.com/rom-rb/rom-factory (if you use rom).

/cc @cllns

**Comment by rosenfeld** [comment]
Hi @davydovanton, thanks for the pointer. Indeed I replaced it with Fabrication just an hour after creating this ticket. If you are curious on the reasons you might want to read my recent articles on the subject:

http://rosenfeld.herokuapp.com/en/articles/ruby-rails/2017-05-01-feeling-alone-in-the-ruby-community-and-replacing-rails-with-roda

And its followup:

http://rosenfeld.herokuapp.com/en/articles/ruby-rails/2017-05-01-feeling-alone-in-the-ruby-community-and-replacing-rails-with-roda

I actually searched about monkey patches in the issues before creating this one. I should probably have searched for ActiveSupport removal instead.

I'm not really bothered by the dependency on ActiveSupport itself. My only problem is that ActiveSupport is patching core Ruby classes, otherwise I wouldn't have a problem with it being a dependency. Other than that I don't have any complaints regarding FactoryGirl. It's a great gem. Fabrication is also another great gem. I like competition and I think FactoryGirl should remain competing in the non Rails Ruby applications field, which do not depend on ActiveSupport themselves. Most of them are probably fighting to remove ActiveSupport due to its monkey patches. I hope the FactoryGirl team would reconsider their decision on keeping the monkey patches as it would be great for the Ruby community to have FactoryGirl as an option for projects avoiding monkey patches.
